curator-framework/src/test/java/org/apache/curator/framework/imps/TestBlockUntilConnected.java

```diff
@@ -232,4 +232,30 @@ public void run()
             CloseableUtils.closeQuietly(client);
         }
     }
+
+    /**
+     * Test that we are actually connected every time that we block until connection is established in a tight loop.
+     */
+    @Test
```

Comment:
qq: how long does this test take to run? not sure it matters, but some projects use @SlowTest or annotations like that